base_local_planner/src/trajectory_planner_ros.cpp

```diff
@@ -166,8 +166,25 @@ namespace base_local_planner {
       private_nh.param("vx_samples", vx_samples, 3);
       private_nh.param("vtheta_samples", vtheta_samples, 20);
 
-      private_nh.param("path_distance_bias", pdist_scale, 0.6);
-      private_nh.param("goal_distance_bias", gdist_scale, 0.8);
+      path_distance_bias = nav_core::loadParameterWithDeprecation(private_nh,
+                                                                  "path_distance_bias",
+                                                                  "pdist_scale",
+                                                                  0.6);
+      goal_distance_bias = nav_core::loadParameterWithDeprecation(private_nh,
+                                                                  "goal_distance_bias",
+                                                                  "gdist_scale",
+                                                                  0.6);
+      // values of the deprecated params need to be applied to the current params, as defaults 
+      // of defined for dynamic reconfigure will override them otherwise.
+      if (private_nh.hasParam("pdist_scale"))
```

Comment:
should this logic actually have "& !private_nh.hasParam("path_distance_bias")" -- I presume we only want to do this once

Reply:
Isn't this executed just once? On node restart the path_distance_bias parameter is deleted if it is not set in the config file, so this needs to be done on every node start. Or do you mean something else?

However, I should add your suggestion for consistency in case of both variables being defined in the config file, because the `loadParameterWithDeprecationWarning` call also favours path_distance_bias over pdist_scale.